packetbeat/protos/tcp/tcp.go

```diff
@@ -404,14 +407,15 @@ type inputMetrics struct {
 
 // newInputMetrics returns an input metric for the TCP processor. If id or
 // device is empty a nil inputMetric is returned.
-func newInputMetrics(id, device string) *inputMetrics {
+func newInputMetrics(id, device string, ports map[uint16]protos.Protocol) *inputMetrics {
 	if id == "" || device == "" {
 		// An empty id signals to not record metrics,
 		// while an empty device means we are reading
 		// from a pcap file and no metrics are needed.
 		return nil
 	}
-	reg, unreg := inputmon.NewInputRegistry("tcp", id+"::"+device, nil)
+	devID := fmt.Sprintf("%s-tcp%s::%s", id, portList(ports), device)
```

Comment:
There is the option to construct a hash for this, but I chose to use the proto and port list since their presence makes the ID immediately informative.

Reply:
The seems fine. The IDs could get quite long for some complex configurations though.

Reply:
If it becomes an issue we can revisit.